SignalShareExtension/ShareViewController.swift

```diff
@@ -315,4 +318,91 @@ public class ShareViewController: UINavigationController, SAELoadViewDelegate {
     public func shareExtensionWasCancelled() {
         self.extensionContext!.completeRequest(returningItems: [], completionHandler: nil)
     }
+
+    // MARK: Helpers
+
+    private func presentConversationPicker() {
+        let conversationPicker = SendExternalFileViewController()
+        buildAttachment().then { attachment -> Void in
+            conversationPicker.attachment = attachment
+            self.pushViewController(conversationPicker, animated: true)
+            Logger.info("presented conversation picker with attachment: \(attachment)")
+        }.catch { error in
+            OWSAlerts.showAlert(withTitle: NSLocalizedString("SHARE_EXTENSION_UNABLE_TO_BUILD_ATTACHMENT_ALERT_TITLE", comment: "Shown when trying to share content to a Signal user for the share extension. Followed by failure details."),
```

Comment:
Option: we could also replace the loading view:

```
   failureTitle = NSLocalizedString("SHARE_EXTENSION_UNABLE_TO_BUILD_ATTACHMENT_ALERT_TITLE", comment: "Shown when trying to share content to a Signal user for the share extension. Followed by failure details.")
   failureMessage = error.localizedDescription
        showErrorView(title:failureTitle, message:failureMessage)
```

Also it seems like this line should have wrapped?

Reply:
I think an alert will work fine for this error case. 

I've extracted the method to be assigned to a local variable to minimize source width.